dl/src/main/scala/com/intel/analytics/bigdl/dataset/DataSet.scala

```diff
@@ -211,16 +216,18 @@ class CachedDistriDataSet[T: ClassTag] private[dataset] (buffer: RDD[Array[T]])
   }).reduce(_ + _)
 
   protected var indexes: RDD[Array[Int]] = buffer.mapPartitions(iter => {
-    Iterator.single(RandomGenerator.shuffle((0 until iter.next().length).toArray))
```

Comment:
In optimizer, there is dataset.shuffle() before training, so old behavior will not be changed.